tests/components/influxdb/test_init.py

```diff
@@ -1111,3 +1158,165 @@ def fast_monotonic():
+            influxdb.DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            _get_write_api_mock_v1,
+            influxdb.DEFAULT_API_VERSION,
+        ),
+        (
+            influxdb.API_VERSION_2,
+            BASE_V2_CONFIG,
+            _get_write_api_mock_v2,
+            influxdb.API_VERSION_2,
+        ),
+    ],
+    indirect=["mock_client", "get_mock_call"],
+)
+async def test_event_listener_attribute_name_conflict(
+    hass, mock_client, config_ext, get_write_api, get_mock_call
+):
+    """Test the event listener when an attribute conflicts with another field."""
+    handler_method = await _setup(hass, mock_client, config_ext, get_write_api)
+
+    attrs = {"value": "value_str"}
+    state = MagicMock(
+        state=1,
+        domain="fake",
+        entity_id="fake.something",
+        object_id="something",
+        attributes=attrs,
+    )
+    event = MagicMock(data={"new_state": state}, time_fired=12345)
+    body = [
+        {
+            "measurement": "fake.something",
+            "tags": {"domain": "fake", "entity_id": "something"},
+            "time": 12345,
+            "fields": {"value": 1, "value__str": "value_str"},
+        }
+    ]
+    handler_method(event)
+    hass.data[influxdb.DOMAIN].block_till_done()
+
+    write_api = get_write_api(mock_client)
+    assert write_api.call_count == 1
+    assert write_api.call_args == get_mock_call(body)
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, get_write_api, get_mock_call, test_exception",
+    [
+        (
+            influxdb.DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            _get_write_api_mock_v1,
+            influxdb.DEFAULT_API_VERSION,
+            ConnectionError("fail"),
+        ),
+        (
+            influxdb.DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            _get_write_api_mock_v1,
+            influxdb.DEFAULT_API_VERSION,
+            influxdb.exceptions.InfluxDBClientError("fail"),
+        ),
+        (
+            influxdb.API_VERSION_2,
+            BASE_V2_CONFIG,
+            _get_write_api_mock_v2,
+            influxdb.API_VERSION_2,
+            ConnectionError("fail"),
+        ),
+        (
+            influxdb.API_VERSION_2,
+            BASE_V2_CONFIG,
+            _get_write_api_mock_v2,
+            influxdb.API_VERSION_2,
+            influxdb.ApiException(),
+        ),
+    ],
+    indirect=["mock_client", "get_mock_call"],
+)
+async def test_connection_failure_on_startup(
+    hass, caplog, mock_client, config_ext, get_write_api, get_mock_call, test_exception
+):
+    """Test the event listener when it fails to connect to Influx on startup."""
+    write_api = get_write_api(mock_client)
+    write_api.side_effect = test_exception
+    config = {"influxdb": config_ext}
+
+    with patch(f"{INFLUX_PATH}.event_helper") as event_helper:
+        assert await async_setup_component(hass, influxdb.DOMAIN, config)
+        await hass.async_block_till_done()
+
+        assert (
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)